claimManagement/src/main/java/org/openimis/imisclaims/domain/entity/PendingClaim.java

```diff
@@ -0,0 +1,423 @@
+        dest.writeString(comment);
+        dest.writeString(total);
+        dest.writeString(icdCode1);
+        dest.writeString(icdCode2);
+        dest.writeString(icdCode3);
+        dest.writeString(icdCode4);
+        dest.writeString(visitType);
+        dest.writeTypedList(services);
+        dest.writeTypedList(medications);
+    }
+
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @NonNull
+    public Date getClaimDate() {
+        return claimDate;
+    }
+
+    @NonNull
+    public String getHealthFacilityCode() {
+        return healthFacilityCode;
+    }
+
+    @Nullable
+    public String getClaimAdminCode() {
+        return claimAdminCode;
+    }
+
+    @NonNull
+    public String getClaimCode() {
+        return claimCode;
+    }
+
+    @Nullable
+    public String getGuaranteeNumber() {
+        return guaranteeNumber;
+    }
+
+    @NonNull
+    public String getChfId() {
+        return chfId;
+    }
+
+    @NonNull
+    public Date getStartDate() {
+        return startDate;
+    }
+
+    @NonNull
+    public Date getEndDate() {
+        return endDate;
+    }
+
+    @NonNull
+    public String getIcdCode() {
+        return icdCode;
+    }
+
+    @Nullable
+    public String getComment() {
+        return comment;
+    }
+
+    @Nullable
+    public String getTotal() {
+        return total;
+    }
+
+    @Nullable
+    public String getIcdCode1() {
+        return icdCode1;
+    }
+
+    @Nullable
+    public String getIcdCode2() {
+        return icdCode2;
+    }
+
+    @Nullable
+    public String getIcdCode3() {
+        return icdCode3;
+    }
+
+    @Nullable
+    public String getIcdCode4() {
+        return icdCode4;
+    }
+
+    @NonNull
+    public String getVisitType() {
+        return visitType;
+    }
+
+    @NonNull
+    public List<Service> getServices() {
+        return services;
+    }
+
+    @NonNull
+    public List<Medication> getMedications() {
+        return medications;
+    }
+
+    public static final Creator<PendingClaim> CREATOR = new Creator<>() {
+        @Override
+        public PendingClaim createFromParcel(Parcel in) {
+            return new PendingClaim(in);
+        }
+
+        @Override
+        public PendingClaim[] newArray(int size) {
+            return new PendingClaim[size];
+        }
+    };
+
+    public static class Service implements Parcelable {
+        @NonNull
+        public static List<Service> fromJson(@NonNull JSONArray array) throws Exception {
+            List<Service> list = new ArrayList<>();
+            for (int i = 0; i < array.length(); i++) {
+                JSONObject medication = array.getJSONObject(i);
+                list.add(new Service(
+                        /* code = */ medication.getString("ServiceCode"),
+                        /* price = */ Double.parseDouble(medication.getString("ServicePrice")),
+                        /* quantity = */ Integer.parseInt(medication.getString("ServiceQuantity"))
```

Comment:
as we're dealing with `Service`:

```suggestion
                JSONObject service = array.getJSONObject(i);
                list.add(new Service(
                        /* code = */ service.getString("ServiceCode"),
                        /* price = */ Double.parseDouble(service.getString("ServicePrice")),
                        /* quantity = */ Integer.parseInt(service.getString("ServiceQuantity"))
```

Reply:
Copypasta gone wrong :sweat_smile: